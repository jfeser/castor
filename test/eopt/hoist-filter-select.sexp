((queries
  ("select([a, b], filter(a = b, atuple([ascalar(0 as a), ascalar(0 as b)], cross)))"
   "select([a], filter(a = b, atuple([ascalar(0 as a), ascalar(0 as b)], cross)))"))
 (transforms (hoist-filter-select))
 (extract_well_staged))
