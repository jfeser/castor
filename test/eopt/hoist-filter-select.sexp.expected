$4:
  select([a, b], filter((a = b), atuple([ascalar(0 as a), ascalar(0 as b)], cross)))
  filter((a = b), select([a, b], atuple([ascalar(0 as a), ascalar(0 as b)], cross)))
  
$5:
  select([a], filter((a = b), atuple([ascalar(0 as a), ascalar(0 as b)], cross)))
  
